### ISSUE TITLE
ImportError: cannot import name 'init_std_stream_encoding' from 'conda.common.compat'

### ISSUE THREAD

**Issue by Syzygianinfern0** [mamba-org/mamba#1706]
My mamba install was working fine till now when I suddenly get this today on any mamba command. 

```
❯ mamba
Traceback (most recent call last):
  File "/home/user/miniconda3/bin/mamba", line 11, in <module>
    sys.exit(main())
  File "/home/user/miniconda3/lib/python3.9/site-packages/mamba/mamba.py", line 892, in main
    from conda.common.compat import ensure_text_type, init_std_stream_encoding
ImportError: cannot import name 'init_std_stream_encoding' from 'conda.common.compat' (/home/user/miniconda3/lib/python3.9/site-packages/conda/common/compat.py)
```

**Comment by wolfv** [comment]
Hey, 

the latest versions of mamba do not use this anymore. It was removed in this commit: https://github.com/mamba-org/mamba/commit/6b614e35681d56f5907068b6db5c54a77d81f3c4

Is it possible for you to update mamba?

**Comment by singletS** [comment]
I had this problem, too, so I updated mamba using conda:
```
#Name                    Version                   Build  Channel
libmamba                  0.23.3               hd8a31e3_1    conda-forge
libmambapy                0.23.3           py39hd55135b_1    conda-forge
mamba                     0.23.3           py39hfa8f2c8_1    conda-forge
```
Now, mamba no longer gives the "cannot import name 'init_std_stream_encoding'" error, but it fails silently when running anything other than "mamba --version".

If it matters, we're running SMP Debian 5.10.113-1 (2022-04-29) x86_64 GNU/Linux

**Comment by Dwingkak** [comment]
I have encountered this problem two days ago. I found that downgrading conda version from 4.13.0 to 4.12.0 can solve my problem.

**Comment by Syzygianinfern0** [comment]
> Hey,
> 
> the latest versions of mamba do not use this anymore. It was removed in this commit: [6b614e3](https://github.com/mamba-org/mamba/commit/6b614e35681d56f5907068b6db5c54a77d81f3c4)
> 
> Is it possible for you to update mamba?

@wolfv 
```
❯ conda update --all
Collecting package metadata (current_repodata.json): done
Solving environment: done

# All requested packages already installed.
```

**Comment by Syzygianinfern0** [comment]
> I had this problem, too, so I updated mamba using conda:
> 
> ```
> #Name                    Version                   Build  Channel
> libmamba                  0.23.3               hd8a31e3_1    conda-forge
> libmambapy                0.23.3           py39hd55135b_1    conda-forge
> mamba                     0.23.3           py39hfa8f2c8_1    conda-forge
> ```
> 
> Now, mamba no longer gives the "cannot import name 'init_std_stream_encoding'" error, but it fails silently when running anything other than "mamba --version".
> 
> If it matters, we're running SMP Debian 5.10.113-1 (2022-04-29) x86_64 GNU/Linux

I can confirm this exact thing after I ran `conda update mamba -c conda-forge`. @wolfv

**Comment by Lash-360** [comment]
I am experiencing the same issue

> Traceback (most recent call last):
  File "C:\Users\Fejiro\anaconda3\Scripts\mamba-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Users\Fejiro\anaconda3\lib\site-packages\mamba\mamba.py", line 848, in main
    from conda.common.compat import ensure_text_type, init_std_stream_encoding
ImportError: cannot import name 'init_std_stream_encoding' from 'conda.common.compat' (C:\Users\Fejiro\anaconda3\lib\site-packages\conda\common\compat.py)

I have tried updating but I still have the same result.

**Comment by Direct-A** [comment]
I have a fully new install. And I have the same problem.

```plaintext
❯ mamba env list
Traceback (most recent call last):
  File "/home/xxx/miniforge3/bin/mamba", line 11, in <module>
    sys.exit(main())
  File "/home/xxx/miniforge3/lib/python3.9/site-packages/mamba/mamba.py", line 892, in main
    from conda.common.compat import ensure_text_type, init_std_stream_encoding
ImportError: cannot import name 'init_std_stream_encoding' from 'conda.common.compat' (/home/xxx/miniforge3/lib/python3.9/site-packages/conda/common/compat.py)
```

**Comment by maximlt** [comment]
We experienced the same error on our CI yesterday, here's one example: https://github.com/holoviz/hvplot/runs/6644190608?check_suite_focus=true

**Comment by wolfv** [comment]
@maximlt you are getting a pretty old version of `mamba` (0.15.4). 

You are also heavily mixing "defaults" and "conda-forge" which is not guaranteed to work (and conda-forge doesn't currently ship the conda `4.13` release).

Would it be possible for you to switch to conda-forge entirely?

**Comment by maximlt** [comment]
Oh yes indeed that is an old version!

I wasn't aware mamba was not guaranteed to work when installed in a base env whose packages come from the default channel, maybe that should be added in the [installation docs](https://github.com/mamba-org/mamba#installation).

I wonder why I got such an old version, but anyway I understand now why things are breaking, thanks for the pointers.

**Comment by Syzygianinfern0** [comment]
> @maximlt you are getting a pretty old version of `mamba` (0.15.4).
> 
> You are also heavily mixing "defaults" and "conda-forge" which is not guaranteed to work (and conda-forge doesn't currently ship the conda `4.13` release).
> 
> Would it be possible for you to switch to conda-forge entirely?

Could you please check my conda env? 

https://github.com/mamba-org/mamba/issues/1706#issuecomment-1140361144

**Comment by wolfv** [comment]
@Syzygianinfern0 you have the same old version of mamba, and the most recent conda version which breaks most mamba versions.

The previous conda and all mamba versions work fine. (conda 4.12 + mamba 0.23.3 for example).

**Comment by yehoshuadimarsky** [comment]
Thanks, I had the same issue, fixed it by bumping it in my Dockerfile

```diff
-FROM quay.io/condaforge/mambaforge:4.10.3-3 AS conda-layer
+FROM quay.io/condaforge/mambaforge:4.12.0-0 AS conda-layer
```

**Comment by ozancaglayan** [comment]
Yesterday i issued a `mamba update --all` in my mambaforge installation and got exactly the issue in the https://github.com/mamba-org/mamba/issues/1706#issuecomment-1140328378 e.g. `mamba --version` prints the version but no other subcommand yields any output.

A few mins ago I completely removed my mambaforge installation, redownloaded it from the `latest` link in GitHub.
After the installation finished, the command was working. I then did `mamba update --all` again, which updated conda from 4.12 to 4.13. Now I'm getting the error mentioned by the op:

```
(base) [olala] ~ $ conda --version
conda 4.13.0
(base) [olala] ~ $ mamba
Traceback (most recent call last):
  File "/home/ozan/mambaforge/bin/mamba", line 11, in <module>
    sys.exit(main())
  File "/home/ozan/mambaforge/lib/python3.9/site-packages/mamba/mamba.py", line 914, in main
    from conda.common.compat import ensure_text_type, init_std_stream_encoding
ImportError: cannot import name 'init_std_stream_encoding' from 'conda.common.compat' (/home/ozan/mambaforge/lib/python3.9/site-packages/conda/common/compat.py)
```

I am not mixing anaconda with mamba and directly using mambaforge for a more clean experience but still, I don'T see how to fix this issue.

**Comment by jpivarski** [comment]
[conda-forge now has conda 4.13.0](https://anaconda.org/conda-forge/conda) (as of 2 days ago), and so a fresh installation of MambaForge followed by

```bash
mamba update --all
```

breaks mamba. For the time being, I'll avoid updating the `conda` package, but this breaks a common installation procedure, one that we [recommend for new users](https://scikit-hep.org/user/installing-conda) because of its robustness.

**Comment by hmaarrfk** [comment]
I think there might be something wrong with the repodata patch:
https://github.com/conda-forge/mamba-feedstock/issues/156

@jpivarski  I was able to recreate your issue today too.

**Comment by QuaternionsRock** [comment]
I am also experiencing this issue. I keep my base environment stock (i.e., I don't have any packages installed beyond what mambaforge comes with). I ran `mamba update --all` today and mamba is completely broken.

**Comment by zwets** [comment]
> For anyone stumbling upon this issue looking for a quick fix like I did, running `conda install conda=4.12` in your base environment should do the trick for the time being.

This worked for me only after `conda remove mamba`. Then, when I tried to reinstall mamba (`conda install mamba -c conda-forge`), I saw conda pulling in 4.13.0 again :-/

The trick then was `conda install mamba conda=4.12.0 -c conda-forge`.

**Comment by jpivarski** [comment]
As @wolfv pointed out to me at SciPy, this issue has been resolved. I just had a chance to test it and it's true:

```bash
mamba update --all
```

leaves mamba in a working state. It updates conda to 4.13.0, which has a change that broke mamba 0.23.3, but the latest mamba is 0.25.0, which works fine with the new conda.

(This is great because I can wholeheartedly recommend [mambaforge](https://github.com/conda-forge/miniforge#mambaforge) to Python beginners again.)

**Comment by jpivarski** [comment]
I forgot to say it, but I meant to say that this issue can be closed.

**Comment by Syzygianinfern0** [comment]
> I forgot to say it, but I meant to say that this issue can be closed.

Sure! Thanks for the fix ❤️

**Comment by sorenwacker** [comment]
conda uninstall mamba -y && conda clean -afy && conda install -c conda-forge mamba -y 

fixed it for me.

**Comment by jpivarski** [comment]
It's fixed in the sense that the latest `conda` package no longer breaks the latest `mamba`. I think if you start with old versions and pass through this knot again, the same thing would happen again, but there's nothing that can be done about that—old versions are old.

I said (in https://github.com/mamba-org/mamba/issues/1706#issuecomment-1208204724) that the problem has been fixed because I started a new mambaforge past the bad dates, allowed `conda` to update to its latest version, and the bug did not materialize. (Contrary to https://github.com/mamba-org/mamba/issues/1706#issuecomment-1208206416, I didn't fix the bug; I just noticed that someone in the conda team had and passed on that knowledge here.)

If you encounter some issue starting with the latest versions of mambaforge and conda, then that would probably be a new issue.

**Comment by duhd1993** [comment]
OK. Thanks for explanation. Still not sure how I ended up in that situation cuz I didn't specifically update conda alone. I was able to downgrade conda to 4.12 and update mamba using mamba. Now everything works. I tried to remove mamba using conda, but got stuck in solving environment.
